components/public-api-server/pkg/apiv1/editor_service.go

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License.AGPL.txt in the project root for license information.
+
+package apiv1
+
+import (
+	"context"
+	"sort"
+
+	connect "github.com/bufbuild/connect-go"
+	"github.com/gitpod-io/gitpod/common-go/log"
+	v1 "github.com/gitpod-io/gitpod/components/public-api/go/experimental/v1"
+	"github.com/gitpod-io/gitpod/components/public-api/go/experimental/v1/v1connect"
+	protocol "github.com/gitpod-io/gitpod/gitpod-protocol"
+	"github.com/gitpod-io/gitpod/public-api-server/pkg/proxy"
+)
+
+func NewEditorService(pool proxy.ServerConnectionPool) *EditorService {
+	return &EditorService{
+		connectionPool: pool,
+	}
+}
+
+var _ v1connect.EditorServiceHandler = (*EditorService)(nil)
+
+type EditorService struct {
+	connectionPool proxy.ServerConnectionPool
+
+	v1connect.UnimplementedEditorServiceHandler
+}
+
+func (s *EditorService) ListEditorOptions(ctx context.Context, req *connect.Request[v1.ListEditorOptionsRequest]) (*connect.Response[v1.ListEditorOptionsResponse], error) {
```

Comment:
do we need some happy path tests for it similar to other services?

Reply:
Good point about adding tests, will add them.

Reply:
Donezies ✅